lambdas/revoke-vcs/src/main/java/uk/gov/di/ipv/core/revokevcs/RevokeVcsHandler.java

```diff
@@ -0,0 +1,204 @@
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            String.format("Revoking VC (%s / %s)", i + 1, numberOfVcs)));
+
+            try {
+                revoke(userIdCriIdPair, i, numberOfVcs);
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Successfully revoked VC (%s / %s)", i + 1, numberOfVcs)));
+            } catch (Exception e) {
+                LOGGER.error(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Unexpected error occurred (%s / %s): %s",
+                                        i + 1, numberOfVcs, e.getMessage())));
+                sendRevokedFailureAuditEvent(
+                        userIdCriIdPair.getUserId(), e.getMessage(), i, numberOfVcs);
+            }
+        }
+    }
+
+    private void revoke(UserIdCriIdPair userIdCriIdPair, int i, int numberOfVcs) throws Exception {
+        // Read VC with userId and CriId
+        var vcStoreItem =
+                verifiableCredentialService.getVcStoreItem(
+                        userIdCriIdPair.getUserId(), userIdCriIdPair.getCriId());
+
+        if (vcStoreItem != null) {
+            // Archive VC
+            archivedVcDataStore.create(vcStoreItem);
+
+            // Submit audit event
+            sendVcRevokedAuditEvent(userIdCriIdPair.getUserId(), vcStoreItem, i, numberOfVcs);
+
+            // Delete VC from the main table
+            verifiableCredentialService.deleteVcStoreItem(
+                    userIdCriIdPair.getUserId(), userIdCriIdPair.getCriId());
+        } else {
+            throw new RevokeVcException("VC cannot be found");
+        }
+    }
+
+    private void sendVcRevokedAuditEvent(
+            String userId, VcStoreItem vcStoreItem, int i, int numberOfVcs)
+            throws ParseException, SqsException, JsonProcessingException {
+        var auditEventUser = new AuditEventUser(userId, null, null, null);
+
+        var signedCredential = SignedJWT.parse(vcStoreItem.getCredential());
+        var jwtClaimsSet = signedCredential.getJWTClaimsSet();
+        var vc = (JSONObject) jwtClaimsSet.getClaim("vc");
+        var evidence = vc.getAsString("evidence");
+
+        var auditExtensions = new AuditExtensionsVcEvidence(jwtClaimsSet.getIssuer(), evidence);
+        var auditEvent =
+                new AuditEvent(
+                        AuditEventTypes.IPV_VC_REVOKED,
+                        configService.getSsmParameter(ConfigurationVariable.COMPONENT_ID),
+                        auditEventUser,
+                        auditExtensions);
+        LOGGER.debug(
+                LogHelper.buildLogMessage(
+                        String.format(
+                                "Sending audit event IPV_VC_REVOKED (%s / %s)",
+                                i + 1, numberOfVcs)));
```

Comment:
We can probably remove this once you're happy. If you need, I think there's a dummy lambda that logs audit events in build (and we can get the QAs to help test it properly in staging).